lib/database_rewinder.rb

```diff
@@ -82,16 +82,28 @@ def all_table_names(connection)
       end
     end
 
+    def get_config_from(connection_name)
+      if Gem::Version.new(Rails.version) >= Gem::Version.new("6.0.0")
+        database_configuration.configs_for(env_name: connection_name).first.configuration_hash
+      else
+        database_configuration[connection_name]
+      end
+    end
+
     def get_cache_key(connection_pool)
       if connection_pool.respond_to?(:db_config) # ActiveRecord >= 6.1
-        connection_pool.db_config.config
+        if Gem::Version.new(Rails.version) >= Gem::Version.new("6.0.0")
```

Comment:
Do we need this check? The line above already checks for ActiveRecord >= 6.1. In any case, `configuration_hash` is only available on Rails 6.1.